components/claim-mgt/org.wso2.carbon.claim.mgt/src/main/java/org/wso2/carbon/claim/mgt/ClaimManagerHandler.java

```diff
@@ -50,6 +58,327 @@ public static ClaimManagerHandler getInstance() {
         return INSTANCE;
     }
 
+    /**
+     * Returns all supported claims.
+     *
+     * @return
+     * @throws ClaimManagementException
+     */
+    public Claim[] getAllSupportedClaims() throws ClaimManagementException {
+        ClaimManager claimManager;
+
+        try {
+            UserRealm realm = getRealm();
+            claimManager = realm.getClaimManager();
+            if (claimManager != null) {
+                // There can be cases - we get a request for an external user
+                // store - where we don'
+                // have a claims administrator.
+                ClaimMapping[] mappings = realm.getClaimManager()
+                        .getAllSupportClaimMappingsByDefault();
+                Claim[] claims = new Claim[mappings.length];
+                for (ClaimMapping mapping : mappings) {
+                    claims[1] = mapping.getClaim();
```

Comment:
Yes, its does not make sense to have like this, even the original commit\[1] have this
 
It should have claims array iterated.

[1] https://github.com/darshanasbg/carbon-identity-framework/blob/de510362e3ef42845293e803a0c0060db3fe1864/components/claim-mgt/org.wso2.carbon.claim.mgt/src/main/java/org/wso2/carbon/claim/mgt/ClaimManagerHandler.java#L77